pkg/sidecar/sidecar.go

```diff
@@ -92,22 +92,16 @@ func Run(runnerName string, resultPath string) error {
 			// Now let the test case tell us how to configure the network.
 			subtree := sync.NetworkSubtree(instance.Hostname)
 			networkChanges := make(chan *sync.NetworkConfig, 16)
-			closeSub, err := instance.Watcher.Subscribe(subtree, networkChanges)
-			if err != nil {
+			if _, err := instance.Watcher.Subscribe(ctx, subtree, networkChanges); err != nil {
```

Comment:
Shouldn't we have a `closeSub` in a `defer` statement?

Reply:
Doesn't matter now. The subscription will be canceled when the context is canceled.

We only return that function (a) for convenience and (b) to indicate to the user that the subscription needs to be released somehow.

Thoughts? Would it be cleaner to remove it? Should we just always call it?

Reply:
I only raised this, because I saw we used to call it - previously cancel was called when this function returns, and now it is called when the context is canceled and I am not sure if this is the same time, but it looks like it is the same time, so as you said - doesn't matter.